File: pallycon-drm-sdk/pallycon_drm_sdk_ios/ios/Classes/models/EventMessage.swift
//
//  EventMessage.swift
//  connectivity_plus
//
//  Created by sungju Yun on 2023/01/04.
//

import Foundation

class EventMessage {
    var contentId: String
    var url: String
    var eventType: PallyConEventType
    var message: String
    var errorCode: String
    
    init(contentId: String, url: String, eventType: PallyConEventType, message: String, errorCode: String) {
        self.contentId = contentId
        self.url = url
        self.eventType = eventType
        self.message = message
        self.errorCode = errorCode
    }
    
    public func toMap() -> Dictionary<String, String> {
        var event = [String: String]()
        
        event["eventType"] = eventType.name
        event["contentId"] = contentId
        event["url"] = url

        if (!errorCode.isEmpty) {
            event["errorCode"] = errorCode
        }

        if (!message.isEmpty) {
            event["message"] = message
        }
        
        return event
    }
}

File: pallycon-drm-sdk/pallycon_drm_sdk_android/android/src/main/kotlin/com/pallycon/pallycon_drm_sdk_android/models/EventMessage.kt
package com.pallycon.pallycon_drm_sdk_android.models

import com.google.gson.Gson
import java.util.HashMap

data class EventMessage(
    val eventType: EventType,
    val contentId: String,
    val url: String,
    val message: String = "",
    val errorCode: String = ""
) {
    fun toJson(): String {
        return Gson().toJson(this)
    }

    fun toMap(): MutableMap<String, Any> {
        val event: MutableMap<String, Any> = HashMap()
        event["eventType"] = eventType.toString()
        event["contentId"] = contentId
        event["url"] = url

        if (errorCode.isNotEmpty()) {
            event["errorCode"] = errorCode
        }

        if (message.isNotEmpty()) {
            event["message"] = message
        }

        return event
    }
}
